rust/arrow/src/array/ffi.rs

```diff
@@ -55,11 +64,45 @@ impl TryFrom<ArrayData> for ffi::ArrowArray {
     type Error = ArrowError;
 
     fn try_from(value: ArrayData) -> Result<Self> {
+        // If parent is nullable, then children also must be nullable
+        // so we pass this nullable to the creation of hte child data
+        let nullable = match value.data_type() {
+            DataType::List(field) => field.is_nullable(),
+            DataType::LargeList(field) => field.is_nullable(),
+            _ => false,
+        };
+
         let len = value.len();
         let offset = value.offset() as usize;
         let null_count = value.null_count();
         let buffers = value.buffers().to_vec();
         let null_buffer = value.null_buffer().cloned();
+        let child_data = value
+            .child_data()
+            .iter()
+            .map(|arr| {
+                let len = arr.len();
```

Comment:
This is because I otherwise cannot pass `nullable: bool` information from the parent. If should split this up in a function separate from `try_from` to make this more explicit.

Reply:
Well, you'll need to handle recursive types more generally anyway. Think `list(list(int8))`.

Reply:
Yes, that would not work like this. I will fix that.